src/python/pants/backend/nfpm/fields/all.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from enum import Enum
+from typing import ClassVar
+
+from pants.engine.target import StringField
+from pants.util.strutil import help_text
+
+
+class NfpmPackageNameField(StringField):
+    nfpm_alias = "name"
+    alias: ClassVar[str] = "package_name"
+    required = True
+    help = help_text(
+        """
+        The package name.
+        """
+    )
+
+
+class GoArch(Enum):
+    # GOARCH possible values come from `okgoarch` var at:
+    # https://github.com/golang/go/blob/go1.20.3/src/cmd/dist/build.go#L62-L79
+    _386 = "386"
+    amd64 = "amd64"
+    arm = "arm"
+    arm64 = "arm64"
+    loong64 = "loong64"
+    mips = "mips"
+    mipsle = "mipsle"
+    mips64 = "mips64"
+    mips64le = "mips64le"
+    ppc64 = "ppc64"
+    ppc64le = "ppc64le"
+    riscv64 = "riscv64"
+    s390x = "s390x"
+    sparc64 = "sparc64"
+    wasm = "wasm"
+
+
+class GoOS(Enum):
+    # GOOS possible values come from `okgoos` var at:
+    # https://github.com/golang/go/blob/go1.20.3/src/cmd/dist/build.go#L81-L98
+    # TODO: maybe filter this down to only what nFPM can handle
```

Comment:
I'm torn. On one hand it would be bad to list a value that we know is going to fail, otoh a newer version of nFPM might add support for a value and then the user is blocked until a new release of this backend is out adding it..